docs/08-extensions.md

```diff
@@ -0,0 +1,129 @@
+
+Deploy an instance of Couchbase by clicking the "Create" button in the top left. The YAML editor has been pre-filled with a set of minimal defaults for our instance.
+
+Operators may require certain input from the developer, in this case a Secret containing the default set of credentials for the database. First, [follow the instructrions](https://docs.couchbase.com/operator/1.1/couchbase-cluster-config.html#authsecret) to generate a Secret or use the OpenShift UI to create a "key/value" Secret with both `username` and `password`. Here's an example secret with username `couchbase` and password `password`.
+
+```
+kind: Secret
+apiVersion: v1
+metadata:
+  name: cb-example-auth
+  namespace: default
+data:
+  password: cGFzc3dvcmQ=
+  username: Y291Y2hiYXNl
+type: Opaque
+```
+
+### Changing Couchbase Parameters
+
+After your Secret is created, refer to it's name with the `authSecret` parameter:
+
+```
+apiVersion: couchbase.com/v1
+kind: CouchbaseCluster
+metadata:
+  name: cb-example
+  namespace: default
+spec:
+  authSecret: cb-example-auth
+  baseImage: registry.connect.redhat.com/couchbase/server
+  buckets:
+    - conflictResolution: seqno
+      enableFlush: true
+      evictionPolicy: fullEviction
+      ioPriority: high
+      memoryQuota: 128
+      name: default
+      replicas: 3
+      type: couchbase
+  ...
+```
+
+Set the `replicas` field set to `3`, so our Operator sets up a highly available cluster for us.
+
+Click "Create". Afterwards, you will be taken to a list of all Couchbase instances running with this Project.
+
+### View the Deployed Resources
+
+Navigate to the Couchbase Cluster that was deployed, and click on the "Resources" tab. This collects all of the objects deployed and managed by the Operator. From here you can view Pod logs to check on the Couchbase Cluster instances.
+
+We are going to use the Service `<insert me>` to access the Couchbase dashboard. Use the OpenShift Console or the CLI to create a Route that points to this Service. Be aware that we are temporarily exposing this dashboard to the Internet for ease of use.
+
+```
+kind: Route
+apiVersion: route.openshift.io/v1
+metadata:
+  name: couchbase-public
+  namespace: default
+  labels:
+    <insert these>
+spec:
+  to:
+    kind: Service
+    name: <service-name>
+    weight: 100
+  port:
+    targetPort: http
+```
+
+The Couchbase dashboard is now available at `couchbase-public-default.apps.<cluster-name>.<base-domain>`.
+
+### Re-Configure the Cluster with the Operator
+
+Keep the dashboard up as we re-configure the cluster. As the Operator scales up more Pods, they will automatically join and appear in the dashboard.
+
+Edit your `cb-example` Couchbase instance to have `4` replicas instead of `3`. A few things will happen:
```

Comment:
Here is a little confusing, actually, the `size` field controls the pod number, not the above `replicas`. As below:
```console
apiVersion: couchbase.com/v1
kind: CouchbaseCluster
metadata:
  creationTimestamp: '2019-03-07T05:51:45Z'
  generation: 1
  name: cb-example
  namespace: default
  resourceVersion: '175458'
  selfLink: /apis/couchbase.com/v1/namespaces/default/couchbaseclusters/cb-example
  uid: 1735101f-409d-11e9-9c32-0a5a70137f08
spec:
  authSecret: cb-example-auth
  baseImage: registry.connect.redhat.com/couchbase/server
  buckets:
    - conflictResolution: seqno
      enableFlush: true
      evictionPolicy: fullEviction
      ioPriority: high
      memoryQuota: 128
      name: default
      replicas: 1
      type: couchbase
  cluster:
    autoFailoverTimeout: 120
    autoFailoverOnDataDiskIssues: true
    clusterName: cb-example
    indexStorageSetting: memory_optimized
    analyticsServiceMemoryQuota: 1024
    eventingServiceMemoryQuota: 256
    searchServiceMemoryQuota: 256
    autoFailoverMaxCount: 3
    dataServiceMemoryQuota: 256
    autoFailoverOnDataDiskIssuesTimePeriod: 120
    indexServiceMemoryQuota: 256
    autoFailoverServerGroup: false
  exposeAdminConsole: false
  servers:
    - name: all_services
      services:
        - data
        - index
        - query
        - search
        - eventing
        - analytics
      size: 4
  softwareUpdateNotifications: false
  version: 5.5.3-3
```